### PR TITLE
[MDS-425] Switch v2

### PR DESCRIPTION
**Pull request by tkullisaar**

### CODE REVIEW

File: lib/moon_web/examples/design/switch_example/using_with_forms.ex [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  alias Moon.Components.Form
+  alias MoonWeb.Pages.Tutorials.AddDataUsingForm.User
+
+  prop(user_changeset, :any, default: User.changeset(%User{}))
+
+  def render(assigns) do
+    ~F"""
+    <div>
+      <Form for={@user_changeset}>
+        <Switch id="switch_11" />
```
Comment (phgrey) [inline lib/moon_web/examples/design/switch_example/using_with_forms.ex:17]: name and value should be set here. name should be a required property for Moon.Design.Form.Switch

File: lib/moon_web/components/themes_select/rtl_switcher.ex [hunk 2]
```diff
@@ -3,34 +3,18 @@ defmodule MoonWeb.Components.ThemesSelect.RtlSwitcher do
 
   use Moon.StatelessComponent
```
Comment (phgrey) [inline lib/moon_web/components/themes_select/rtl_switcher.ex:4]: do we still need this component at all?

File: lib/moon_web/components/themes_select/theme_switcher.ex [hunk 3]
```diff
@@ -3,24 +3,20 @@ defmodule MoonWeb.Components.ThemesSelect.ThemeSwitcher do
 
   use Moon.StatelessComponent
 
-  alias Moon.Components.Form
-  alias Moon.Design.Switch
-  alias Moon.Components.Field
-  alias MoonWeb.Components.ThemesSelect.SelectedTheme
+  alias Moon.Icons.OtherMoon
```
Comment (phgrey) [inline lib/moon_web/components/themes_select/theme_switcher.ex:6]: again - do we still need this component at all?
is it used anywhere?

File: lib/moon/design/form/switch.ex [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  def render(assigns) do
+    ~F"""
+    <div>
+      <input
+        type="checkbox"
+        name={@name}
+        value={@value}
+        checked={@is_switched}
+        class="hidden"
+        :on-click="toggle_switch"
```
Comment (phgrey) [inline lib/moon/design/form/switch.ex:35]: why do we need this event here?

File: lib/moon/design/switch.ex [hunk 5]
```diff
@@ -1,118 +1,92 @@
 defmodule Moon.Design.Switch do
   @moduledoc false
 
-  use Moon.StatelessComponent
-  alias Moon.Icon
-  alias Moon.Components.Switch.Caption
-  alias Phoenix.LiveView.JS
+  use Moon.StatefulComponent
```
Comment (phgrey) [inline lib/moon/design/switch.ex:4]: any reasons for making it statefull?

File: lib/moon_web/examples/design/switch_example/for_full_customization.ex [hunk 6]
```diff
@@ -0,0 +1,182 @@
+defmodule MoonWeb.Examples.Design.SwitchExample.CodeExampleForFullCustomization do
```
Comment (phgrey) [inline lib/moon_web/examples/design/switch_example/for_full_customization.ex:1]: i do not see any Switch usage here. what is this example at all?

**Review by phgrey — COMMENTED** [review]
take a look at the comments please

File: lib/moon_web/examples/design/switch_example/using_with_forms.ex (hunk 1 above)
Comment (tkullisaar) [reply lib/moon_web/examples/design/switch_example/using_with_forms.ex:17]: Missing attributes added

File: lib/moon_web/components/themes_select/rtl_switcher.ex (hunk 2 above)
Comment (tkullisaar) [reply lib/moon_web/components/themes_select/rtl_switcher.ex:4]: Not needed. I had deleted this component, but probably reverted the change by accident and didn't notice.

File: lib/moon_web/components/themes_select/theme_switcher.ex (hunk 3 above)
Comment (tkullisaar) [reply lib/moon_web/components/themes_select/theme_switcher.ex:6]: Not needed. I had deleted this component, but probably reverted the change by accident and didn't notice.

File: lib/moon/design/form/switch.ex (hunk 4 above)
Comment (tkullisaar) [reply lib/moon/design/form/switch.ex:35]: This event toggles the checkbox. Should it be done in some other way?

File: lib/moon_web/examples/design/switch_example/for_full_customization.ex (hunk 6 above)
Comment (tkullisaar) [reply lib/moon_web/examples/design/switch_example/for_full_customization.ex:1]: My mistake, misunderstood the example. Please have a look at the new example. However - the positioning of the switcher is not yet custamisable and I would like to ask advice on what would be the best way to solve it.

File: lib/moon/design/form/switch.ex (hunk 4 above)
Comment (phgrey) [reply lib/moon/design/form/switch.ex:35]: but this checkbox is hidden, so the event will never occur as i understand

**Review by phgrey — APPROVED** [review]

**Pull request merged**